src/theory/quantifiers/bv_inverter.cpp

```diff
@@ -261,24 +262,114 @@ Node BvInverter::getPathToPv(Node lit, Node pv, Node sv, Node pvs,
   return slit;
 }
 
-Node BvInverter::solve_bv_constraint(Node sv,
-                                     Node sv_t,
-                                     Node t,
-                                     Kind rk,
-                                     bool pol,
-                                     std::vector<unsigned>& path,
-                                     BvInverterModelQuery* m,
-                                     BvInverterStatus& status) {
-  unsigned index;
-  unsigned nchildren;
+Node BvInverter::solve_bv_lit(Node sv,
+                              Node lit,
+                              bool pol,
+                              std::vector<unsigned>& path,
+                              BvInverterModelQuery* m,
+                              BvInverterStatus& status) {
+  Assert(!path.empty());
+
   NodeManager* nm = NodeManager::currentNM();
+  unsigned index, nchildren;
+  Kind k;
+
+  index = path.back();
+  Assert(index < lit.getNumChildren());
+  path.pop_back();
+  k = lit.getKind();
+
+  if (k == NOT) {
+    pol = !pol;
+    lit = lit[index];
+  }
+
+  Assert(k == EQUAL
+       || k == BITVECTOR_COMP
+       || k == BITVECTOR_ULT
+       || k == BITVECTOR_ULTBV
+       || k == BITVECTOR_SLT
+       || k == BITVECTOR_SLTBV);
+
+  Assert((k != EQUAL && k != BITVECTOR_COMP) || pol == true);
```

Comment:
why does this hold?

Reply:
As discussed offline, this does not hold for disequalities of course, but these we also didn't handle previously. I'll add handling for disequalities.